File: C++/bee 1040.cpp
#include <iostream>
#include <iomanip>
#include <cmath>

using namespace std;

int main() {
    float n1, n2, n3, n4;
    cin >> n1 >> n2 >> n3 >> n4;

    float media = (n1 * 2 + n2 * 3 + n3 * 4 + n4) / 10;
    cout << fixed << setprecision(1) << "Media: " << media << endl;
    if (media >= 7) {
        cout << "Aluno aprovado." << endl;
    } else if (media < 5) {
        cout << "Aluno reprovado." << endl;
    } else if (media >= 5 && media <= 6.9){
        cout << "Aluno em exame." << endl;
        float n5;
        cin >> n5;
        cout << "Nota do exame: " << n5 << endl;
        if ((n5 + media) / 2 >= 5) {
            cout << fixed << setprecision(1) << "Aluno aprovado." << endl << "Media final: " << (n5 + media) / 2 << endl;
        } else {
            cout << fixed << setprecision(1) << "Aluno reprovado." << endl << "Media final: " << (n5 + media) / 2 << endl;
        }

    }

    return 0;
}

File: Java/bee 1040.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        double n1, n2, n3, n4;
        n1 = sc.nextDouble();
        n2 = sc.nextDouble();
        n3 = sc.nextDouble();
        n4 = sc.nextDouble();

        double media = (n1 * 2 + n2 * 3 + n3 * 4 + n4 * 1)/10.0;
        media = Math.floor(media * 10)/10.0;
        System.out.printf("Media: %.1f\n", media);
        if(media >= 7){
            System.out.println("Aluno aprovado.");
        } else if(media < 5){
            System.out.println("Aluno reprovado.");
        } else{
            System.out.println("Aluno em exame.");
            double n5 = sc.nextDouble();
            double mediaFinal = (n5 + media)/2;
            System.out.printf("Nota do exame: %.1f\n", n5);
            if(mediaFinal >= 5){
                System.out.println("Aluno aprovado.");
            } else{
                System.out.println("Aluno reprovado.");
            }
            System.out.printf("Media final: %.1f\n", mediaFinal);

        }
    }
}
